File: 02._data_formats/homework/parser.py
# Reading text files
def read_text_file(file_path):
    with open(file_path, 'r') as file:
        return file.read()

# Parsing XML files
import xmltodict

def parse_xml_file(file_path):
    with open(file_path, 'r') as file:
        xml_content = file.read()
        return xmltodict.parse(xml_content)

# Parsing YAML files
import yaml

def parse_yaml_file(file_path):
    with open(file_path, 'r') as file:
        return yaml.safe_load(file)

# Parsing JSON files
import json

def parse_json_file(file_path):
    with open(file_path, 'r') as file:
        return json.load(file)

# Parsing CSV files
import pandas as pd

def parse_csv_file(file_path):
    return pd.read_csv(file_path, header=0).squeeze(0).to_dict()

#Data parsed to dictionary (except for txt)
json_data = parse_json_file('me.json')
xml_data = parse_xml_file('me.xml')
csv_data = parse_csv_file('me.csv')
txt_data = read_text_file('me.txt')
yaml_data = parse_yaml_file('me.yaml')

File: 02._data_formats/homework/parser.js
// JavaScript script for reading and parsing files in Node.js using npm packages

// Import the built-in `fs` module for reading files
import fs from 'fs';

// Import the `fast-xml-parser` package for parsing XML files
import { XMLParser } from 'fast-xml-parser';

// Import the `yamljs` package for parsing YAML files
import yaml from 'yamljs';

// Import the `csvtojson` package for parsing CSV files
import csv from 'csvtojson';

// Reading text files
function readTextFile(filePath) {
    return fs.readFileSync(filePath, 'utf-8');
  };
  
  // Parsing XML files
  function parseXmlFile(filePath) {
    const xml = fs.readFileSync(filePath, 'utf-8');
    let parser = new XMLParser()
    return parser.parse(xml).me
  };
  
  // Parsing YAML files
  function parseYamlFile(filePath) {
    const yamlData = readTextFile(filePath);
    return yaml.parse(yamlData);
  };
  
  // Parsing JSON files
  function parseJsonFile(filePath) {
    const json = readTextFile(filePath);
    return JSON.parse(json);
  };

  // Parsing CSV files
function parseCsvFile(filePath) {
    return csv().fromFile(filePath);
  };

export {readTextFile, parseXmlFile, parseYamlFile, parseJsonFile, parseCsvFile}